AirCasting/SDSync/MobileSessions/SDSyncAveragingService.swift

```diff
@@ -0,0 +1,77 @@
+// Created by Lunar on 29/09/2022.
+//
+
+import Foundation
+import CoreLocation
+
+protocol SDSyncAveragingService {
+    func averageMeasurementsWithReminder<T: AverageableMeasurement>(measurements: [T], startTime: Date, averagingWindow: AveragingWindow, averagingAction: (T, [T]) -> Void) -> [T]
+}
+
+protocol AverageableMeasurement {
+    var time: Date! { get set }
+    var value: Double { get set }
+    var location: CLLocationCoordinate2D? { get set }
+}
+extension MeasurementEntity: AverageableMeasurement { }
```

Comment:
we're gonna need to add a `var measurementTime: Date` here in that case. It will just return `date`:
```Swift
var measurementTime: Date { date }
```